### ISSUE TITLE
Frequency Domain filtering in scikit-image 

### ISSUE THREAD

**Issue by vighneshbirodkar** [scikit-image/scikit-image#1618]
I'd like to contribute frequency domain filters to skimage. I want to know if this is a feature we'd like to have and how would we go about it. My intial guess is the code would live inside the `filters` module. 

Some of the things I'd like to include would be
* ideal, butterworth and gaussian filters
* handle zero padding implicitly 
* helper functions to compute the fourier transform and back 
* helper function to visualize frequency domain plots

**Comment by glemaitre** [comment]
Hey,

I am preparing a lecture for an MSc course (Introduction to Image Processing) based closely on Gonzalez and Wood book. In the same time, I would like to rely on `skimage` regarding the practise. Since that some filters are missing this is the opportunity for to contribute as well for this part.

Let me know if you advance on the question.

Cheers,

**Comment by vighneshbirodkar** [comment]
Sometime back I did speak to @stefanv about this and we weren't able to figure out how to define a consistent API for these filters. It is still a pending issue I would say.

Meanwhile,  you can check out
http://fredo-editor.github.io
It's a little tool I made to visualize frequency domaim filtering.

**Comment by glemaitre** [comment]
Thanks for the tool.

Regarding the API, do you have any log of the discussion somewhere?

**Comment by vighneshbirodkar** [comment]
No, it was on a Skype call and I don't quite remember. @stefanv Could you chip in ?

**Comment by grlee77** [comment]
This was partly completed by #5382 (Butterworth)
